src/OculusDriverNode.cpp

```diff
@@ -0,0 +1,214 @@
+    pub_imgUniform = this->create_publisher<sensor_msgs::msg::Image>("image_uniform", 10);
+    pub_imgCartesian = this->create_publisher<sensor_msgs::msg::Image>("image_cartesian", 10);
+    pub_pressure = this->create_publisher<sensor_msgs::msg::FluidPressure>("pressure", 10);
+    pub_temperature = this->create_publisher<sensor_msgs::msg::Temperature>("temperature", 10);
+    pub_orientation = this->create_publisher<geometry_msgs::msg::Vector3Stamped>("orientation", 10);
+    pub_configuration = this->create_publisher<sonar_driver_interfaces::msg::SonarConfiguration>("configuration", 10);
+    pub_bearings = this->create_publisher<sonar_driver_interfaces::msg::SonarBearings>("bearings", 10);
+    
+    // cv_imgShared_ = cv_bridge::CvImage(std_msgs::msg::Header(), sensor_msgs::image_encodings::MONO8, cv::Mat(512, 512, CV_8U));
+    sonar_ = std::make_unique<OculusSonar>(cvBridgeShared_);
+
+    UniformBearingCorrectorConfig initConfig(0, 0, 0.0, 0.0, 0.0);
+    bearingCorrector_ = std::make_shared<UniformBearingCorrector>(initConfig);
+
+    updateCommonHeader();
+    cvBridgeUniform_->encoding = "mono8";
+    cvBridgeCartesian_->encoding = "mono8";
+
+    sub_reconfiguration = this->create_subscription<sonar_driver_interfaces::msg::SonarConfigurationChange>(
+        "reconfigure", 10, std::bind(&OculusDriverNode::cb_reconfiguration, this, std::placeholders::_1)
+    );
+
+}
+
+/// @brief Method to execute upon receival of a simplePingResult in the sonar_ class 
+/// @param image 
+void OculusDriverNode::cb_simplePingResult(std::unique_ptr<SonarImage>& image){
+    updateCommonHeader();
+    printf("OculusDriverNode: Correcting image\n");
+
+    double angularResolution = (sonar_->getBearingTable().back() - sonar_->getBearingTable().front()) / (100.0 * image->width);
+    // spdlog::info("back: {}. front: {}. Diff: {}. Angular: {}. Width: {}", sonar_->getBearingTable().back(), sonar_->getBearingTable().front(), (sonar_->getBearingTable().back() - sonar_->getBearingTable().front()), angularResolution, image->width);
+    UniformBearingCorrectorConfig currentConfig(image->height, image->width, 
+                                                sonar_->getMinimumRange(), sonar_->getMaximumRange(),
+                                                angularResolution);
+
+    if(!bearingCorrector_->hasSameConfig(currentConfig)){
+        currentConfig.setBearings(sonar_->getBearingTable());
+        bearingCorrector_ = std::make_shared<UniformBearingCorrector>(currentConfig);
+    }
+
+
+    bearingCorrector_->rectifyImage(cvBridgeShared_->image, cvBridgeUniform_->image, cvBridgeCartesian_->image);
+    
+    printf("OculusDriverNode: Corrected image\n");
+    
+    publishImage();
+    printf("OculusDriverNode: Published image\n");
+    publishUniformImage();
+    printf("OculusDriverNode: Published Uniform\n");
+    publishCartesianImage();
+    printf("OculusDriverNode: Published Cartesian\n");
+    publishCurrentConfig();
```

Comment:
Maybe we should remove these prints to reduce spam and improve performance as the sonar can sample with up to 40Hz.